src/client/src/ui/spotTile/model/spotTileModel.js

```diff
@@ -113,22 +114,29 @@ export default class SpotTileModel extends ModelBase {
       let request = this._createTradeRequest(e.direction);
       this._log.info(`Will execute ${request.toString()}`);
       this.isTradeExecutionInFlight = true;
+      window.clearTimeout(this.notificationTimer);
       this._executionDisposable.setDisposable(
-        this._executionService.executeTrade(request).subscribeWithRouter(
-          this.router,
-          this.modelId,
-          (response:ExecuteTradeResponse) => {
-            this.isTradeExecutionInFlight = false;
-            this.notification = response.hasError
-              ? TradeExecutionNotification.createForError(response.error)
-              : TradeExecutionNotification.createForSuccess(response.trade);
-          },
-          err => {
-            this.isTradeExecutionInFlight = false;
-            this._log.error(`Error executing ${request.toString()}. ${err}`, err);
-            this.notification = TradeExecutionNotification.createForError(`Unknown stream error`);
+        this._executionService.executeTrade(request)
+          .do(response => {
```

Comment:
not a massive fan of side effects in a `do` method, additionally it's setting local state which is a bit dodgy.  

I think we should move this into the bode of the observer (i.e. the function passed to the subscribe function).